src/sidebar/oauth-auth.js

```diff
@@ -79,14 +85,20 @@ function auth($http, settings) {
 
   // Set a timeout to refresh the access token a few minutes before it expires.
   function refreshAccessTokenBeforeItExpires(tokenInfo) {
-    var delay = tokenInfo.expiresIn * 1000;
-
     // We actually have to refresh the access token _before_ it expires.
     // If the access token expires in one hour, this should refresh it in
     // about 55 mins.
-    delay = Math.floor(delay * 0.91);
+    var refreshAfter = tokenInfo.expiresAt.valueOf() * 0.91;
+
+    function refreshAccessTokenIfNearExpiry() {
+      if (Date.now() > refreshAfter) {
+        refreshAccessToken(tokenInfo.refreshToken);
+      } else {
+        refreshAccessTokenBeforeItExpires(tokenInfo);
+      }
+    }
 
-    window.setTimeout(refreshAccessToken, delay, tokenInfo.refreshToken);
+    window.setTimeout(refreshAccessTokenIfNearExpiry, 5000);
```

Comment:
So instead of changing `setTimeout` to `setInterval` I changed it to multiple calls to `setTimeout` each with a delay of 5 secs (rather than the previous one call with a delay of 55 mins)